### PR TITLE
Changes in OrderOnTransfer

### PR DESCRIPTION
**Pull request by GNUp**

### CODE REVIEW

**Comment by GNUp** [comment]
This PR now fail at `mint-and-transfer-with-order` test since the order transaction is now disabled currently.

**Comment by majecty** [comment]
You can use the Order transaction when running a CodeChain with "ENABLE_ORDER" environment variable.

**Comment by GNUp** [comment]
I checked that `mint-and-transfer-with-order` works but I don't know how to make the travis pass

**Comment by GNUp** [comment]
`mint-and-transfer-with-order` test fails until https://github.com/CodeChain-io/codechain/pull/1589#issue-283740315 this PR is merged.

**Comment by majecty** [comment]
CodeChain and the SDK cannot be updated at the same time.
Please follow the steps below.

1. Create a branch in your own SDK repository.
2. Use the SDK from your repository in the CodeChain integration tests.
3. After merging your PR in CodeChain, use the updated version of CodeChain in this PR.
4. Merge this PR.
5. Use the latest SDK in CodeChain.

**Comment by GNUp** [comment]
This PR can be merged after https://github.com/CodeChain-io/codechain/pull/1589#issue-283740315

**Comment by majecty** [comment]
@joojis Could you review this PR?

**Comment by joojis** [comment]
@majecty Is there a spec that I can refer to? I don't know what the fields mean.

**Comment by majecty** [comment]
The original "inputIndices" and "outputIndices" are subdivided.
I sent a PR to update CodeChain spec.
https://github.com/CodeChain-io/codechain-sdk-js/pull/385#issuecomment-510868153

File: src/core/transaction/TransferAsset.ts [hunk 1]
```diff
@@ -215,21 +215,46 @@ export class TransferAsset extends Transaction implements AssetTransaction {
...
+            spentQuantity,
+            inputFromIndices = [],
+            inputFeeIndices = [],
+            outputFromIndices = [],
+            outputToIndices = [],
+            outputOwnedFeeIndices = [],
+            outputTransferredFeeIndices = []
+        } = params;
+        if (inputFromIndices.length === 0) {
             throw Error(`inputIndices should not be empty`);
```
Comment (joojis) [inline src/core/transaction/TransferAsset.ts]: Shouldn't it be changed to `inputFromIndices`?

File: src/key/index.ts [hunk 2]
```diff
@@ -305,7 +305,10 @@ export class Key {
         if (tx instanceof TransferAsset) {
             let flag = false;
             for (const order of tx.orders()) {
-                if (order.inputIndices.indexOf(index) !== -1) {
+                const inputindices = order.inputFromIndices.concat(
```
Comment (joojis) [inline src/key/index.ts]: Why not camelCase here?

**Pull request merged**